### PR TITLE
Fix csv.py parser mappings

### LINKED ISSUE #5601: CSV Read Fails on dtypes.int32 After Recent int_ Removal
The recent removal of `dtypes.int_` caused an error in a benchmark.  When switching to `dtypes.int32` instead, the benchmark still fails but with a different error. I've attached the stack trace.  

It looks like the CSV read does not support all the scalar dtypes.  I'm looking for a true 32 bit int, not a 64 bit one, for my test.  (Though I'm not sure that's doable from the python side.  Maybe np.int32?)  Exception is raised from `/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/csv.py`.

[csv-read-datatype-exception.txt](https://github.com/user-attachments/files/15792949/csv-read-datatype-exception.txt)

### PR DESCRIPTION
**Pull request by devinrsmith**
Fixes #5601

### CODE REVIEW

**Comment by devinrsmith** [comment]
My earlier assessment that there was a bug in the lower layer is incorrect; I was using the wrong column names.

File: py/server/deephaven/csv.py [hunk 1]
```diff
@@ -75,9 +75,9 @@ def read(
                 dht.byte: _JParsers.BYTE,
                 dht.char: _JParsers.CHAR,
                 dht.short: _JParsers.SHORT,
-                dht.int64: _JParsers.INT,
```
Comment (stanbrub) [inline py/server/deephaven/csv.py:78]: This disables support for `dht.int64` and `dht.float64`.  That seems more accurate, but how many users will it affect?

File: py/server/deephaven/csv.py [hunk 2]
```diff
@@ -75,9 +75,9 @@ def read(
...
+                dht.int32: _JParsers.INT,
                 dht.long: _JParsers.LONG,
```
Comment (rcaudy) [inline py/server/deephaven/csv.py:79]: Do int64 and float64 correctly map to long and double?

File: py/server/deephaven/csv.py (hunk 1 above)
Comment (devinrsmith) [reply py/server/deephaven/csv.py:78]: `dht.int64 == dht.long`, `dht.float64 == dht.double`; they are present lower down in the map.

File: py/server/deephaven/csv.py (hunk 2 above)
Comment (devinrsmith) [reply py/server/deephaven/csv.py:79]: yes, they are the same objects.

File: py/server/deephaven/csv.py (hunk 1 above)
Comment (stanbrub) [reply py/server/deephaven/csv.py:78]: I'm afraid I don't see dht.int64, dht.float64, dht.int16 or dht.int8 anywhere in this file.  What file is that in?

**Review by jmao-denver — APPROVED** [review]
LGTM

File: py/server/deephaven/csv.py (hunk 1 above)
Comment (stanbrub) [reply py/server/deephaven/csv.py:78]: I think I see it.  This is an object map, so you only need one variable that points to the object to be represented.  So in `dtypes.py`, both int64 and long point to the same DType object, so you only need one represented in the map.

**Review by stanbrub — APPROVED** [review]

**Pull request merged**